pkg/broker/handler/pool/fanout/pool.go

```diff
@@ -33,23 +33,31 @@ import (
 	"github.com/google/knative-gcp/pkg/broker/handler/processors/deliver"
 	"github.com/google/knative-gcp/pkg/broker/handler/processors/fanout"
 	"github.com/google/knative-gcp/pkg/broker/handler/processors/filter"
+	"github.com/google/knative-gcp/pkg/utils"
 )
 
 // SyncPool is the sync pool for fanout handlers.
 // For each broker in the config, it will attempt to create a handler.
 // It will also stop/delete the handler if the corresponding broker is deleted
 // in the config.
 type SyncPool struct {
-	options *pool.Options
-	targets config.ReadonlyTargets
-	pool    sync.Map
+	options   *pool.Options
+	targets   config.ReadonlyTargets
+	pool      sync.Map
+	projectID string
 }
 
 // StartSyncPool starts the sync pool.
 func StartSyncPool(ctx context.Context, targets config.ReadonlyTargets, opts ...pool.Option) (*SyncPool, error) {
+	options := pool.NewOptions(opts...)
+	projectID, err := utils.ProjectID(options.ProjectID)
+	if err != nil {
+		return nil, err
+	}
 	p := &SyncPool{
-		targets: targets,
-		options: pool.NewOptions(opts...),
+		targets:   targets,
+		options:   options,
+		projectID: projectID,
```

Comment:
There is already an option for project id: https://github.com/google/knative-gcp/blob/master/pkg/broker/handler/pool/options.go#L95-L99